tango/common/util.py

```diff
@@ -153,3 +159,16 @@ def could_be_class_name(name: str) -> bool:
 
 def _is_valid_python_name(name: str) -> bool:
     return bool(name and name[0].isalpha() and name.isalnum())
+
+
+def find_open_port() -> int:
```

Comment:
Borrowed from AllenNLP, so this is actually pretty well tested.

Reply:
The problem is that this is not thread safe. Two tango processes starting up at the same time might still attempt to start on the same port, because there is a gap between `find_open_port()` and binding to the port. This could be more likely than you think if someone starts tango processes from a script, which would have them go through startup at exactly the same time.

Can you just try ports when the server starts up, the way the web server does?